src/server/keystore/keystore.js

```diff
@@ -0,0 +1,103 @@
+import { writeFileSync, readFileSync, existsSync } from 'fs';
+import { createCipheriv, createDecipheriv, randomBytes, pbkdf2Sync } from 'crypto';
+import * as errors from './errors';
+
+const VERSION = 1;
+const ALGORITHM = 'aes-256-gcm';
+const ITERATIONS = 250;
+
+export class Keystore {
+  constructor(path, password = '') {
+    this.path = path;
+    this.data = {};
+    this.password = password;
+
+    this.load();
+  }
+
+  static errors = errors;
+
+  static encrypt(text, password = '') {
+    const iv = randomBytes(12);
+    const salt = randomBytes(64);
+    const key = pbkdf2Sync(password, salt, ITERATIONS, 32, 'sha512');
+
+    const cipher = createCipheriv(ALGORITHM, key, iv);
+
+    const ciphertext = Buffer.concat([cipher.update(text, 'utf8'), cipher.final()]);
+    const tag = cipher.getAuthTag();
+
+    return Buffer.concat([salt, iv, tag, ciphertext]).toString('base64');
+  }
+
+  static decrypt(data, password = '') {
+    try {
+      const bData = new Buffer(data, 'base64');
+
+      // convert data to buffers
+      const salt = bData.slice(0, 64);
```

Comment:
thought on  using constants here  and with the generation above?  i think it would make it a little easier to understand where the sizes are coming from.